frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx

```diff
@@ -93,13 +95,19 @@ class PublicOrEmbeddedDashboardInner extends Component<PublicOrEmbeddedDashboard
       setErrorPage,
       parameterQueryParams,
       dashboardId,
+      isNavigatingBackToDashboard,
     } = this.props;
 
-    initialize();
+    const shouldReloadDashboardData = !isNavigatingBackToDashboard;
+
+    initialize({ clearCache: shouldReloadDashboardData });
```

Comment:
I find we still need to pass `forceUpdate` like you did in the previous version. Take this scenario.
- Drill a dashboard
- Navigate back to the dashboard
- Navigate to another dashboard without unmounting

Here `isNavigatingBackToDashboard` will always be `true` whether or not we dispatch `reset` anywhere. I've tried dispatching `reset` inside the hook in the comment above. And before we call `_initialize` in the `componentDidUpdate` life cycle. Inside `_initialize` you would still get `isNavigatingBackToDashboard: true` even though `reset` was already dispatched.

I suspect this happens because somehow Redux updates the state asynchronously. I don't know why, I thought by default it should be synchronous, since `reset` is a sync action. Maybe that has something to do with our redux setup, but I wouldn't go that far.

This cause a problem when i'm on a dashboard with an empty tab.
- drill 
- navigate back
- click on an empty tab
- navigate to another dashboard without unmounting

My dashboard won't be loaded.